syne_tune/optimizer/schedulers/searchers/random_searcher.py

```diff
@@ -0,0 +1,38 @@
+import logging
+from typing import Optional, List, Dict, Any
+
+from syne_tune.optimizer.schedulers.searchers.searcher import BaseSearcher
+
+logger = logging.getLogger(__name__)
+
+
+class RandomSearcher(BaseSearcher):
+    """
+    Searcher which randomly samples configurations to try next.
+    """
+
+    def __init__(
+        self,
+        config_space: Dict[str, Any],
+        points_to_evaluate: Optional[List[dict]] = None,
```

Comment:
With recent enough python, we can do directly:
```suggestion
        points_to_evaluate: list[dict] | None = None,
```
which avoids the import of optional and list, I would suggest to start doing this as it makes the code much leaner.

Reply:
Rethinking about this one, we probably want to use the old typing in this PR as you did to avoid mixing different thing and having a very large diff.